saviSimulation/savi/jason_processing/UxVBehavior.java

```diff
@@ -128,8 +128,8 @@ public void update(UxV uxv, double simTime, int perceptionDistance, List<WorldOb
             	double dist = polar.get(Geometry.DISTANCE);
             	if (isObjectDetected (azimuth, elevation, dist, perceptionDistance) ) {
 					//it's visible 
-					detectedObjects.add(new CameraPerception(wo.type, this.time, azimuth, elevation, dist, wo.pixels/2));
-					uncoveredObjects.add(new CameraPerception(wo.type, this.time, azimuth, elevation, dist, wo.pixels/2));
+					detectedObjects.add(new CameraPerception(wo.type, wo.ID, this.time, azimuth, elevation, dist, wo.pixels/2));
+					uncoveredObjects.add(new CameraPerception(wo.type, wo.ID, this.time, azimuth, elevation, dist, wo.pixels/2));
```

Comment:
Adding object identifiers to the perception

Reply:
See comment above. I think this is too strong an assumption. Agents needing to communicate about particular objects could rather identify them by their location (threat at location x,y rather than threat id#6).

Reply:
I agree with @ntartania 
For example, in the case study of "chasing birds", If you have a bird at time "R" in a location x,y, and then at time "Q" you also have a bird there, you do not know if it is the same one or not.  That is way you know the "type" (i.e. tree, house, threat, etc.) but not the id. If you want to have different types of threats that is possible

Reply:
Agree with this as well. The ID was more or less thrown in there just to get the proper behaviour going. Will change this, thank you!